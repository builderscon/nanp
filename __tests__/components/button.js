/* eslint-env jest */

import React from 'react'
import Button from '../../src/components/button'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Button
      title='button sample'
      onPress={() => {}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with buttonStyle', () => {
  const tree = renderer.create(
    <Button
      title='button sample'
      onPress={() => {}}
      buttonStyle={{
        flex: 1
      }}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with titleStyle', () => {
  const tree = renderer.create(
    <Button
      title='button sample'
      onPress={() => {}}
      titleStyle={{
        color: 'black'
      }}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
