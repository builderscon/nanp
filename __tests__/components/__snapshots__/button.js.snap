exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}>
      button sample
    </Text>
  </View>
</View>
`;

exports[`test renders correctly with buttonStyle 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}>
      button sample
    </Text>
  </View>
</View>
`;

exports[`test renders correctly with titleStyle 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
        }
      }>
      button sample
    </Text>
  </View>
</View>
`;
